contracts/Manager.sol

```diff
@@ -179,523 +191,308 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
-        uint256 numberGroups = activeGroups.length();
-        uint256 numberVotableGroups = 0;
-        address[] memory votableGroups = new address[](numberGroups);
+        uint256 overflowingCelo = protocolScheduledVotes > fromReceivableVotesByElection
+            ? protocolScheduledVotes - fromReceivableVotesByElection
+            : 0;
 
-        for (uint256 i = 0; i < numberGroups; i++) {
-            address group = activeGroups.at(i);
-            uint256 scheduledVotes = account.scheduledVotesForGroup(group);
-            if (getElection().canReceiveVotes(group, votes + scheduledVotes)) {
-                votableGroups[numberVotableGroups] = group;
-                numberVotableGroups++;
-            }
+        if (overflowingCelo == 0) {
+            revert FromGroupNotOverflowing(fromGroup);
         }
 
-        address[] memory votableGroupsFinal = new address[](numberVotableGroups);
-        for (uint256 i = 0; i < numberVotableGroups; i++) {
-            votableGroupsFinal[i] = votableGroups[i];
+        uint256 toReceivableVotes = getReceivableVotesForGroup(toGroup);
+        if (toReceivableVotes == 0) {
+            revert ToGroupOverflowing(toGroup);
         }
 
-        return votableGroupsFinal;
+        uint256 toMove = Math.min(overflowingCelo, toReceivableVotes);
+        scheduleRebalanceTransfer(fromGroup, toGroup, toMove);
     }
 
     /**
-     * @notice Sorts an array of GroupWithVotes structs based on increasing
-     * `votes` values.
-     * @param groupsWithVotes The array to sort.
-     * @dev This is an in-place insertion sort. In general in Solidity we should
-     * be careful of algorithms on arrays, especially O(n^2) ones, but here
-     * we're guaranteed to be working with a small array, its length is bounded
-     * by the maximum number of groups that can be voted for in Elections.sol.
-     */
-    function sortGroupsWithVotes(GroupWithVotes[] memory groupsWithVotes) internal pure {
-        for (uint256 i = 1; i < groupsWithVotes.length; i++) {
-            uint256 j = i;
-            while (j > 0 && groupsWithVotes[j].votes < groupsWithVotes[j - 1].votes) {
-                (groupsWithVotes[j], groupsWithVotes[j - 1]) = (
-                    groupsWithVotes[j - 1],
-                    groupsWithVotes[j]
-                );
-                j--;
-            }
-        }
+     * @notice Allows strategy to initiate transfer without any checks.
+     * This method is supposed to be used for transfers between groups
+     * only within strategy
+     * @param fromGroups The groups the deposited CELO is intended to be revoked from.
+     * @param fromVotes The amount of CELO scheduled to be revoked from each respective group.
+     * @param toGroups The groups the transferred CELO is intended to vote for.
+     * @param toVotes The amount of CELO to schedule for each respective `toGroups`.
+     */
+    function scheduleTransferWithinStrategy(
```

Comment:
I would prefer to leave it as it is since Account has currently no knowledge about strategies. It would mean that we would have to introduce this concept which feels like it is not necessary.